src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/energy/EnergyNet.java

```diff
@@ -106,10 +103,10 @@ public void onClassificationChange(Location l, NetworkComponent from, NetworkCom
                     consumers.put(l, component);
                     break;
                 case GENERATOR:
-                    if (component instanceof EnergyNetProvider) {
-                        generators.put(l, (EnergyNetProvider) component);
-                    } else if (component instanceof SlimefunItem) {
-                        ((SlimefunItem) component).warn("This Item is marked as a GENERATOR but does not implement the interface EnergyNetProvider!");
+                    if (component instanceof EnergyNetProvider energyNetProvider) {
+                        generators.put(l, energyNetProvider);
+                    } else if (component instanceof SlimefunItem slimefunItem) {
+                        slimefunItem.warn("This Item is marked as a GENERATOR but does not implement the interface EnergyNetProvider!");
```

Comment:
We do not always have to be so literal about it.
```suggestion
                    if (component instanceof EnergyNetProvider provider) {
                        generators.put(l, provider);
                    } else if (component instanceof SlimefunItem item) {
                        item.warn("This Item is marked as a GENERATOR but does not implement the interface EnergyNetProvider!");
```